brush-parser/src/error.rs

```diff
@@ -22,6 +22,49 @@ pub enum ParseError {
     },
 }
 
+#[cfg(feature = "fancy-diagnostics")]
```

Comment:
Is there a reason to put this code behind `fancy-diagnostics`? Is there actually any code in this `mod` that depends on `miette`'s fancy feature (as opposed to just using its structs/types)?

Reply:
`fancy-diagnostics` is a feature for brush-parser, miette/fancy is needed for the example that show how to use brush-parser with miette and the standard fancy visualization.